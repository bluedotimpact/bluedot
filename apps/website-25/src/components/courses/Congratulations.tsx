import React from 'react';
import SocialShare from './SocialShare';

type CongratulationsProps = {
  courseTitle: string;
  courseUrl: string;
  referralCode?: string;
  text?: string;
};

const Congratulations: React.FC<CongratulationsProps> = ({
  courseTitle,
  courseUrl,
  referralCode,
  text,
}) => {
  return (
    <div className="congratulations flex flex-col gap-4 container-lined p-4 bg-white items-center">
      <h3 className="congratulations__title text-center">Congratulations on completing {courseTitle}!</h3>
      <p className="congratulations__description text-center">
        Now share your perspective! Those reflections aren't going to achieve much sitting in an exercise box. Share them with your network to get the conversation going.
      </p>
      <SocialShare
        courseUrl={courseUrl}
        referralCode={referralCode}
        text={text}
      />
    </div>
  );
};

export default Congratulations;
