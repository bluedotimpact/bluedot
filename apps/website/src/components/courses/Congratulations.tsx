import React from 'react';
import clsx from 'clsx';
import SocialShare from './SocialShare';
import { H3, P } from '../Text';

type CongratulationsProps = {
  courseTitle: string;
  coursePath: string;
  text?: string;
  className?: string;
};

const Congratulations: React.FC<CongratulationsProps> = ({
  courseTitle,
  coursePath,
  text,
  className,
}) => {
  const socialShareText = text || `ðŸŽ‰ I just completed the ${courseTitle} course from BlueDot Impact! Itâ€™s free, self-paced, and packed with insights. Check it out and sign up with my referral link below:`;

  return (
    <div className={clsx('congratulations flex flex-col gap-4 container-lined p-4 bg-white items-center', className)}>
      <H3 className="congratulations__title text-center">Congratulations on completing {courseTitle}!</H3>
      <P className="congratulations__description text-center">
        Now share your perspective! Those reflections aren't going to achieve much sitting in an exercise box. Share them with your network to get the conversation going.
      </P>
      <SocialShare
        coursePath={coursePath}
        text={socialShareText}
      />
    </div>
  );
};

export default Congratulations;
