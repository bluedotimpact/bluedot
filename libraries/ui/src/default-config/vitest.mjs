/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
