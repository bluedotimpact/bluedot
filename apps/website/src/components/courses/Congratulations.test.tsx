import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import Congratulations from './Congratulations';

describe('Congratulations', () => {
  test('renders default as expected', () => {
    const { container } = render(
      <Congratulations
        courseTitle="Future of AI"
        coursePath="/courses/future-of-ai"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders with custom args', () => {
    const { container } = render(
      <Congratulations
        courseTitle="Future of AI"
        coursePath="/courses/future-of-ai"
        text="This is a custom text I've written for this course!"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
