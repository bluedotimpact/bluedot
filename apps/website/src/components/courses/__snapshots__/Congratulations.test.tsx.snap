// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Congratulations > renders FoAI course with special component 1`] = `
<div>
  <div
    class="congratulations container-lined p-8 bg-white"
  >
    <div
      class="flex flex-col gap-2"
    >
      <p
        class="congratulations__title bluedot-h4 mb-2 text-center"
      >
        Congratulations on completing the Future of AI course!
      </p>
      <p
        class="congratulations__description bluedot-p"
      >
        Want to connect with others taking AI safety seriously? 
        <span
          class="font-semibold"
        >
          Join 3,000+ FoAI course graduates
        </span>
         turning knowledge into action.
      </p>
    </div>
    <div
      class="mt-4 flex justify-start"
    >
      <a
        class="cta-button flex items-center justify-center transition-all duration-200 w-fit whitespace-nowrap cursor-pointer not-prose text-sm px-4 py-3 rounded-sm font-[650] cta-button--primary bg-bluedot-normal link-on-dark"
        href="https://community.bluedot.org"
        tabindex="0"
        target="_blank"
      >
        Join the Community
      </a>
    </div>
  </div>
</div>
`;

exports[`Congratulations > renders default as expected 1`] = `
<div>
  <div
    class="congratulations flex flex-col gap-4 container-lined p-4 bg-white items-center"
  >
    <h3
      class="bluedot-h3 not-prose congratulations__title text-center"
    >
      Congratulations on completing 
      Test Course
      !
    </h3>
    <p
      class="bluedot-p not-prose congratulations__description text-center"
    >
      Now share your perspective! Those reflections aren't going to achieve much sitting in an exercise box. Share them with your network to get the conversation going.
    </p>
    <div
      class="flex justify-center gap-4"
    >
      <a
        aria-label="X (Twitter)"
        class="flex items-center justify-center size-12 text-gray-700 hover:text-gray-900 transition-colors"
        href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A3000%2Fcourses%2Ftest-course%3Futm_source%3Dreferral%26utm_campaign%3Dtwitter&text=%F0%9F%8E%89%20I%20just%20completed%20the%20Test%20Course%20course%20from%20BlueDot%20Impact!%20It's%20free%2C%20self-paced%2C%20and%20packed%20with%20insights.%20Check%20it%20out%3A"
        tabindex="0"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
          />
        </svg>
      </a>
      <a
        aria-label="LinkedIn"
        class="flex items-center justify-center size-12 text-gray-700 hover:text-gray-900 transition-colors"
        href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A3000%2Fcourses%2Ftest-course%3Futm_source%3Dreferral%26utm_campaign%3Dlinkedin&text=%F0%9F%8E%89%20I%20just%20completed%20the%20Test%20Course%20course%20from%20BlueDot%20Impact!%20It's%20free%2C%20self-paced%2C%20and%20packed%20with%20insights.%20Check%20it%20out%3A"
        tabindex="0"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
          />
        </svg>
      </a>
      <a
        aria-label="Facebook"
        class="flex items-center justify-center size-12 text-gray-700 hover:text-gray-900 transition-colors"
        href="https://www.facebook.com/sharer/sharer.php?u=http%3A%2F%2Flocalhost%3A3000%2Fcourses%2Ftest-course%3Futm_source%3Dreferral%26utm_campaign%3Dfacebook"
        tabindex="0"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`Congratulations > renders with custom args 1`] = `
<div>
  <div
    class="congratulations flex flex-col gap-4 container-lined p-4 bg-white items-center"
  >
    <h3
      class="bluedot-h3 not-prose congratulations__title text-center"
    >
      Congratulations on completing 
      Test Course
      !
    </h3>
    <p
      class="bluedot-p not-prose congratulations__description text-center"
    >
      Now share your perspective! Those reflections aren't going to achieve much sitting in an exercise box. Share them with your network to get the conversation going.
    </p>
    <div
      class="flex justify-center gap-4"
    >
      <a
        aria-label="X (Twitter)"
        class="flex items-center justify-center size-12 text-gray-700 hover:text-gray-900 transition-colors"
        href="https://twitter.com/intent/tweet?url=http%3A%2F%2Flocalhost%3A3000%2Fcourses%2Ftest-course%3Futm_source%3Dreferral%26utm_campaign%3Dtwitter&text=This%20is%20a%20custom%20text%20I've%20written%20for%20this%20course!"
        tabindex="0"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
          />
        </svg>
      </a>
      <a
        aria-label="LinkedIn"
        class="flex items-center justify-center size-12 text-gray-700 hover:text-gray-900 transition-colors"
        href="https://www.linkedin.com/shareArticle?mini=true&url=http%3A%2F%2Flocalhost%3A3000%2Fcourses%2Ftest-course%3Futm_source%3Dreferral%26utm_campaign%3Dlinkedin&text=This%20is%20a%20custom%20text%20I've%20written%20for%20this%20course!"
        tabindex="0"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
          />
        </svg>
      </a>
      <a
        aria-label="Facebook"
        class="flex items-center justify-center size-12 text-gray-700 hover:text-gray-900 transition-colors"
        href="https://www.facebook.com/sharer/sharer.php?u=http%3A%2F%2Flocalhost%3A3000%2Fcourses%2Ftest-course%3Futm_source%3Dreferral%26utm_campaign%3Dfacebook"
        tabindex="0"
        target="_blank"
      >
        <svg
          fill="currentColor"
          height="20"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
